compiler/src/dotty/tools/repl/ReplCompiler.scala

```diff
@@ -59,9 +59,9 @@ class ReplCompiler extends Compiler {
             importContext(imp)(using ctx))
         }
 
-        val rootCtx = super.rootContext
-          .withRootImports   // default root imports
-          .withRootImports(RootRef(() => defn.EmptyPackageVal.termRef) :: Nil)
+        val rootCtx = super.rootContext.fresh
+          .setOwner(defn.EmptyPackageClass)
```

Comment:
@odersky this is the crux of the change: we change the owner (from RootPackage), as well as go back to emitting repl snippets in the empty package.  That way there isn't a change in ownership (from the parser inserting the synthetic empty package) which previously made references found, via `findRefRecur`, in the empty package have precedence over the references that come from these root imports or user-defined imports.